### PR TITLE
colors: fix non-existent self in HCYColorGradient

### PR DESCRIPTION
**Pull request by Rainrider**

### CODE REVIEW

**Comment by p3lim** [comment]
This is a breaking change for anyone that uses `oUF:RGBToHCY` or `oUF:HCYtoRGB`.

**Comment by ls-** [comment]
@p3lim, regarding your first point. And? We're about to push 9.0, breaking backward compatibility isn't that big of a deal.

**Comment by Rainrider** [comment]
This would be the case with the renaming too, so I'd just switch that to the proper casing.

**Comment by p3lim** [comment]
Calling the functions by `oUF:` inside the functions doesn't allow overriding per-layout, only on the global oUF object.

**Comment by Rainrider** [comment]
@p3lim 
Do you mean we should disallow the possibility for a per-layout override by using the global oUF object instead of self?

**Comment by Rainrider** [comment]
What would be your solution? To me passing the object itself feels strange (be it oUF or the frame).

File: elements/additionalpower.lua [hunk 1]
```diff
@@ -68,7 +68,7 @@ local function UpdateColor(element, cur, max)
 	if(element.colorClass) then
 		t = parent.colors.class[playerClass]
 	elseif(element.colorSmooth) then
-		r, g, b = parent.ColorGradient(cur, max, unpack(element.smoothGradient or parent.colors.smooth))
+		r, g, b = oUF:ColorGradient(cur, max, unpack(element.smoothGradient or parent.colors.smooth))
```
[the remark this answers is not in the document]
Comment (p3lim) [reply elements/additionalpower.lua]: Revert to using `parent`

File: elements/health.lua [hunk 2]
```diff
@@ -94,7 +94,7 @@ local function UpdateColor(element, unit, cur, max)
 	elseif(element.colorReaction and UnitReaction(unit, 'player')) then
 		t = parent.colors.reaction[UnitReaction(unit, 'player')]
 	elseif(element.colorSmooth) then
-		r, g, b = parent.ColorGradient(cur, max, unpack(element.smoothGradient or parent.colors.smooth))
+		r, g, b = oUF:ColorGradient(cur, max, unpack(element.smoothGradient or parent.colors.smooth))
```
[the remark this answers is not in the document]
Comment (p3lim) [reply elements/health.lua]: Revert to using `parent`

File: elements/power.lua [hunk 3]
```diff
@@ -137,7 +137,7 @@ local function UpdateColor(element, unit, cur, min, max, displayType)
 		t = parent.colors.reaction[UnitReaction(unit, 'player')]
 	elseif(element.colorSmooth) then
 		local adjust = 0 - (min or 0)
-		r, g, b = parent.ColorGradient(cur + adjust, max + adjust, unpack(element.smoothGradient or parent.colors.smooth))
+		r, g, b = oUF:ColorGradient(cur + adjust, max + adjust, unpack(element.smoothGradient or parent.colors.smooth))
```
[the remark this answers is not in the document]
Comment (p3lim) [reply elements/power.lua]: Revert to using `parent`

File: elements/tags.lua [hunk 4]
```diff
@@ -84,7 +84,9 @@ local _ENV = {
 		end
 		return string.format('|cff%02x%02x%02x', r * 255, g * 255, b * 255)
 	end,
-	ColorGradient = oUF.ColorGradient,
+	ColorGradient = function(...)
+		oUF:ColorGradient(...)
+	end,
```
[the remark this answers is not in the document]
Comment (p3lim) [reply elements/tags.lua]: We don't have a context for the unit frame object here, so we're SOL right now, but we could probably do something about that.

**Comment by p3lim** [comment]
Added comments above.

File: elements/tags.lua (hunk 4 above)
[the remark this answers is not in the document]
Comment (p3lim) [reply elements/tags.lua]: Actually, we can just add it to the environment like we do with colors on lines 724, 739, 755 and 772 (as of [061e04b](https://github.com/oUF-wow/oUF/blob/061e04b63e5045665c7fe01b7ba591b095925b2d/elements/tags.lua))

[the remark this answers is not in the document]
Comment (p3lim) [reply elements/tags.lua]: I opted for exposing the actual parent in that case, that way tags can also benefit from other per-object overridden oUF globals as well (a typical case would be colors, see the [start of the `Shared` function in my layout](https://github.com/p3lim-wow/oUF_P3lim/blob/12c936cde492b57a5f1294765ea83f40fe965814/oUF_P3lim.lua#L486-L489)).

**Comment by Rainrider** [comment]
@p3lim should we expose `ColorGradient` or the specific ones as well through the unit frames?

**Pull request merged**